amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/fileplans/FilePlanTests.java

```diff
@@ -594,4 +603,178 @@ public void listHolds()
             }
         });
     }
+
+    /**
+     * <pre>
+     * Given that a file plan exists
+     * When rmAdmin user ask the API for roles and relations
+     * It provides list of all roles and relations
+     * </pre>
+     */
+    @Test
+    public void listFilePlanAllRolesAndRelation()
+    {
+        // Call to new API to get the roles and capabilities
+        Object response = getRestAPIFactory().getFilePlansAPI().getFilePlanRolesAndRelation(FILE_PLAN_ALIAS, "");
+        assertStatusCode(OK);
+        Map<?, ?> roleModelEntries = (LinkedHashMap<?, ?>) response;
+        List<?> roleModelList = (ArrayList<?>) roleModelEntries.get("entries");
+        assertEquals(roleModelList.size(), 7);
+        Map<?, ?> adminRoleModelMap = getRoleModelMap(response);
+        assertEquals((String) adminRoleModelMap.get("displayLabel"), ROLE_RM_ADMIN.displayName);
+    }
+
+    /**
+     * <pre>
+     * Given that a file plan exists
+     * When rmAdmin user ask the API for roles and relations with SystemRoles as false
+     * It provides list of all roles and relations excluding SystemRoles
+     * </pre>
+     */
+    @Test
+    public void listFilePlanAllRolesAndRelationExcludeSystemRoles()
+    {
+        String parameters = "where=(systemRoles=false)";
+        // Call to new API to get the roles and capabilities
+        Object response = getRestAPIFactory().getFilePlansAPI().getFilePlanRolesAndRelation(FILE_PLAN_ALIAS, parameters);
+        assertStatusCode(OK);
+        Map<?, ?> roleModelEntries = (LinkedHashMap<?, ?>) response;
+        List<?> roleModelList = (ArrayList<?>) roleModelEntries.get("entries");
+        assertEquals(roleModelList.size(), 5);
```

Comment:
This test only checks the number of roles returned. I think we should check if the endpoint does not return system roles.